src/client/ua_client_connect.c

```diff
@@ -1601,6 +1767,8 @@ closeSessionCallback(UA_Client *client, void *userdata,
 
 UA_StatusCode
 UA_Client_disconnectAsync(UA_Client *client) {
+    closeListeningReverseConnectSocket(client);
```

Comment:
The below async approach will call disconnectSecureChannel eventually.
So closing the reverse socket here is redundant.